extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteMSQTestsHelper.java

```diff
@@ -450,6 +452,13 @@ public void close()
       {
       }
     };
-    return () -> new ReferenceCountingResourceHolder<>(segment, Closer.create());
+    DataSegment dataSegment = DataSegment.builder()
+                                         .dataSource(segmentId.getDataSource())
+                                         .interval(segmentId.getInterval())
+                                         .version(segmentId.getVersion())
+                                         .shardSpec(new LinearShardSpec(0))
+                                         .size(0)
+                                         .build();
+    return () -> new ReferenceCountingResourceHolder<>(new SegmentWithMetadata(dataSegment, segment), Closer.create());
```

Comment:
Shoundn't this null datasegment ?

Reply:
Since this is mocking an actual data segment being loaded, this should have a non null datasegment, right?